test/integration/wheel_slip.cc

```diff
@@ -0,0 +1,526 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <ignition/msgs/entity_factory.pb.h>
+
+#include <ignition/common/Console.hh>
+#include <ignition/math/Pose3.hh>
+#include <ignition/transport/Node.hh>
+
+#include <sdf/Sphere.hh>
+#include <sdf/Cylinder.hh>
+
+#include "ignition/gazebo/components/Collision.hh"
+#include "ignition/gazebo/components/Geometry.hh"
+#include "ignition/gazebo/components/Gravity.hh"
+#include "ignition/gazebo/components/Inertial.hh"
+#include "ignition/gazebo/components/Joint.hh"
+#include "ignition/gazebo/components/JointVelocity.hh"
+#include "ignition/gazebo/components/JointVelocityCmd.hh"
+#include "ignition/gazebo/components/Light.hh"
+#include "ignition/gazebo/components/LinearVelocity.hh"
+#include "ignition/gazebo/components/Link.hh"
+#include "ignition/gazebo/components/Model.hh"
+#include "ignition/gazebo/components/Name.hh"
+#include "ignition/gazebo/components/ParentEntity.hh"
+#include "ignition/gazebo/components/Pose.hh"
+#include "ignition/gazebo/components/SlipComplianceCmd.hh"
+#include "ignition/gazebo/components/World.hh"
+#include "ignition/gazebo/Server.hh"
+#include "ignition/gazebo/SystemLoader.hh"
+#include "ignition/gazebo/test_config.hh"
+
+#include "plugins/MockSystem.hh"
+
+using namespace ignition;
+using namespace gazebo;
+
+/// \brief Test DiffDrive system
+class WheelSlipTest : public ::testing::Test
+{
+  // Documentation inherited
+  protected: void SetUp() override
+             {
+               common::Console::SetVerbosity(4);
+               setenv("IGN_GAZEBO_SYSTEM_PLUGIN_PATH",
+                   (std::string(PROJECT_BINARY_PATH) + "/lib").c_str(), 1);
+             }
+};
+
+class Relay
+{
+  public: Relay()
+          {
+            auto plugin = loader.LoadPlugin("libMockSystem.so",
+                "ignition::gazebo::MockSystem",
+                nullptr);
+            EXPECT_TRUE(plugin.has_value());
+
+            this->systemPtr = plugin.value();
+
+            this->mockSystem = static_cast<MockSystem *>(
+                systemPtr->QueryInterface<System>());
+            EXPECT_NE(nullptr, this->mockSystem);
+          }
+
+  public: Relay &OnPreUpdate(MockSystem::CallbackType _cb)
+          {
+            this->mockSystem->preUpdateCallback = std::move(_cb);
+            return *this;
+          }
+
+  public: Relay &OnUpdate(MockSystem::CallbackType _cb)
+          {
+            this->mockSystem->updateCallback = std::move(_cb);
+            return *this;
+          }
+
+  public: Relay &OnPostUpdate(MockSystem::CallbackTypeConst _cb)
+          {
+            this->mockSystem->postUpdateCallback = std::move(_cb);
+            return *this;
+          }
+
+  public: SystemPluginPtr systemPtr;
+
+  private: SystemLoader loader;
+  private: MockSystem *mockSystem;
+};
+
+TEST_F(WheelSlipTest, TireDrum)
```

Comment:
This test doesn't seem complete (when comparing against the Gazebo classic version). Is that on purpose? If so can you leave a todo?

Reply:
I might need some help with one one, I added a todo for now, but was struggling to make the 1:1 correlation from Gazebo classic's tests to this one.

Reply:
Anything to do here, or can we save this test for later?

Reply:
Yes, I think we can skip this for now since the original test uses a force-torque sensor, which is not supported in ign-gazebo yet.